solutions/USACO dec16-cow-checklist.mdx

```diff
@@ -0,0 +1,88 @@
+---
+id: usaco-670
+source: USACO Gold 2016 December
+title: Cow Checklist
+author: Maggie Liu
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_checklist_gold_dec16.html)
+
+<LanguageSection>
+
+<CPPSection>
+
+**Time Complexity:** $\mathcal{O}(H \cdot G)$
+
+```cpp
+#include <iostream>
+#include <cstdio>
+#include <cmath>
+using namespace std;
+
+int energy(pair<int, int> a, pair<int, int> b)
+{
+	return pow(a.first - b.first, 2) + pow(a.second - b.second, 2);
```

Comment:
Not relevant for this problem, but `pow` wouldn't be accurate enough for `long long`'s. Ex.

```cpp
int main() {
	const ll x = 2039593823;
	cout << (ll)pow(x,2) << "\n";
	cout << x*x << "\n";
	// 4159942962819755520
	// 4159942962819755329
}
```

Would prefer it if you used your own function to square an integer.